### ISSUE TITLE
Apply Page Flags at Document Level

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#4379]
* As a (type of user)
* I want (goal) 
* so that (reason/ why/ value) 

**Assumptions & Scope**	
Feedback from the super user sessions - to be filled out and updated later

What is IN scope?

What is NOT in scope?

**Acceptance Criteria**	

**Scenario 1: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 2: xxxxxx**
* GIVEN ... (you or your condition)
* WHEN ... (what you do) 
* THEN ... (what you see) 

**Scenario 3: xxxxxx**
...

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by m-prodan** [comment]
document level flagging... @lmullane @JHarrietha-AOT @liseandtea @KyEggleston

**Comment by Aman-Hundal** [comment]
Refinement Questions:

- Rule of Thumb for development = the latest change to the page flags will take precedent over the previous changes (ie. if record has page 5 with pageflag of withheld in full, if full disclosure is applied that will change page 5s flag to full disclosure. Same rule applies for page flag consults)
- Can we apply all the page flags available to pages for document level page flagging? (ie. in progress flag, or consults flag?)

FE changes needed

**Comment by m-prodan** [comment]
Matt to include a modal scenario - @JHarrietha-AOT to provide design
Will overwrite all REDACTION flags, consult flags will NOT be overwritten, but could be added (link to #4403) -

**Comment by Aman-Hundal** [comment]
Follow up notes:

- Rule of thumb mentioned above is confirmed.
- Business to add another scenario -> confirmation modal to confirm with user that they will apply and overwrite all page flags if document level page flag selected
- Additional logic -> If a page has a redaction and consult page flag applied (ticket 4403) -> if document level redaction flag applied keep consult and overrride the redaction flag -> if document level consult page flagg applied add the document level page flag to page consult (Consults are never overwritten just added to)
- All page flags should be available at document level besides "Page Left Off"

**Comment by m-prodan** [comment]
cc @JHarrietha-AOT

**Comment by JHarrietha-AOT** [comment]
@m-prodan @liseandtea @KyEggleston This is what I am thinking for the content for this module just to ensure users know they are selecting a whole document instead of an individual page. Let me know what you think of the wording and if you have any changes.

https://jacklyn808742.invisionapp.com/overview/Redaction-cla8feezv003o015h7op82c5c/screens

**Comment by KyEggleston** [comment]
Changing the confirm button verbiage from "Create Redline" to "Confirm" - but may have already identified this and were just reusing asset

The verbiage is tough without making too lengthy - more detail would look like this, but unsure about it being too much info:

"Are you sure you want to flag all pages in the selected document(s) as (Page Flag)?

This will override any existing page flags throughout the entire document (Note: this will **not** override existing consult page flags, which will be retained in addition to the new page flag).

If you have selected to add the consult page flag, this will not override other page flags, and it will be added as an additional page flag.

**Comment by KyEggleston** [comment]
NOTE: this is a prerequisite before we can start working on the phased release stories
